dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/HTMLPageAssetRendered.java

```diff
@@ -30,8 +31,23 @@ public HTMLPageAssetRendered(final Host site,
 
         super(site, template, containers, page, layout, canCreateTemplate, canEditTemplate, viewAs, pageUrlMapper, live);
         this.html = html;
+    }
 
+    public HTMLPageAssetRendered(final Host site,
+                                 final Template template,
+                                 final Collection<? extends ContainerRaw> containers,
+                                 final HTMLPageAssetInfo page,
+                                 final TemplateLayout layout,
+                                 final String html,
+                                 final boolean canCreateTemplate,
+                                 final boolean canEditTemplate,
+                                 final ViewAsPageStatus viewAs,
+                                 final String pageUrlMapper,
+                                 final boolean live,
+                                 final Contentlet contentlet) {
```

Comment:
When classes have constructors with so many parameters they become candidates for Builders. See http://www.informit.com/articles/article.aspx?p=1216151&seqNum=2